core/src/main/java/hudson/security/UserDetailsHelper.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.security;
+
+import org.acegisecurity.userdetails.UserDetails;
+
+import javax.annotation.Nonnull;
+
+public class UserDetailsHelper {
```

Comment:
Should this class be restricted? If not, I think it should be `final` with a `private` constructor as is common with util classes.

Reply:
The class is expected to be public as the `isFullyActive` method could be used by any security realm. 

The `final` / `private` is "common" depending on the project. In Jenkins, in general I would say nope, you can look at `Util` or other classes in the core.
I do not have a really strong opinion about that, but I prefer to keep consistency (to ease future change?) instead of everybody "bringing his own salt".

In SonarQube, we will have a warning, that a static class must have such stuff.

Reply:
I was thinking in general Java practice, not Jenkins specifically, but good point. I've even seen people use interfaces instead which allows for silly things.

Reply:
Yes, especially when you want a singleton.